src/sentry/workflow_engine/endpoints/validators/base/data_condition.py

```diff
@@ -1,30 +1,53 @@
+from typing import Generic, TypeVar
+
 from rest_framework import serializers
-from rest_framework.fields import Field
 
 from sentry.api.serializers.rest_framework import CamelSnakeSerializer
-
-
-class BaseDataConditionValidator(CamelSnakeSerializer):
-    type = serializers.CharField(
-        required=True,
-        max_length=200,
-        help_text="Condition used to compare data value to the stored comparison value",
-    )
-
-    @property
-    def comparison(self) -> Field:
-        raise NotImplementedError
-
-    @property
-    def result(self) -> Field:
-        raise NotImplementedError
-
-    def validate(self, attrs):
-        attrs = super().validate(attrs)
-        return attrs
-
-
-class BaseDataConditionGroupValidator(CamelSnakeSerializer):
-    logic_type = serializers.CharField(required=True)
-    organization_id = serializers.IntegerField(required=True)
-    conditions = BaseDataConditionValidator(many=True)
+from sentry.db.models import Model
+from sentry.utils.registry import NoRegistrationExistsError
+from sentry.workflow_engine.endpoints.validators.utils import (
+    validate_json_primitive,
+    validate_json_schema,
+)
+from sentry.workflow_engine.models.data_condition import CONDITION_OPS, Condition
+from sentry.workflow_engine.registry import condition_handler_registry
+from sentry.workflow_engine.types import DataConditionHandler
+
+T = TypeVar("T", bound=Model)
+
+
+class BaseDataConditionValidator(CamelSnakeSerializer[T], Generic[T]):
+    type = serializers.ChoiceField(choices=[(t.value, t.value) for t in Condition])
+
+    comparison = serializers.JSONField(required=True)
+    condition_result = serializers.JSONField(required=True)
+    # condition_group_id = serializers.IntegerField(required=True)
```

Comment:
fyi, this is on purpose. need to think through the data condition group a little more in another PR then think about how this ties together. I think in general, it's correct that we would just want the ID at this point.